File: Python/shared/sec_template_registry_item.py
# Exact full port of SecTemplateRegistryItem.cs

import xml.etree.ElementTree as ET
from .sec_template_item import SecTemplateItem
from .registry_item_helper import RegistryItemHelper
from .effective_state_sources import EffectiveStateSources

class SecTemplateRegistryItem(SecTemplateItem):
    def __init__(self, node: ET.Element):
        super().__init__()
        self.m_regHelper = RegistryItemHelper(self, "ComputerConfig")
        self.ReadStdElems(node)
        self.ReadStdSecTemplateElems(node)

        line_parts = self.m_sLineItem.split(self.KeyValSeparator())
        key_part = line_parts[0][8:]  # skip "Registry"
        last_slash = key_part.rfind("\\")
        self.m_regHelper.m_sKey = key_part[:last_slash].strip()
        self.m_regHelper.m_sValue = key_part[last_slash + 1:].strip()

        type_and_val = line_parts[1].split(",", 2)
        type_code = type_and_val[0]
        if type_code == "1":
            self.m_regHelper.m_sType = "REG_SZ"
        elif type_code == "2":
            self.m_regHelper.m_sType = "REG_EXPAND_SZ"
        elif type_code == "3":
            self.m_regHelper.m_sType = "REG_BINARY"
        elif type_code == "4":
            self.m_regHelper.m_sType = "REG_DWORD"
        elif type_code == "7":
            self.m_regHelper.m_sType = "REG_MULTI_SZ"
        elif type_code == "11":
            self.m_regHelper.m_sType = "REG_QWORD"
        else:
            self.m_regHelper.m_sType = f"regType {type_code}"

        self.m_sBaselineValue = type_and_val[1].strip('"')

    def SortKey(self) -> str:
        return (self.__class__.__name__ + "!" + self.m_regHelper.m_sKey + "!" + self.m_regHelper.m_sValue).lower()

    def Evaluate(self, sources: EffectiveStateSources, x_output_doc: ET.Element):
        self.m_regHelper.RetrieveRegistryValue(sources.hklm64)  # hklm64 will be set in EffectiveStateSources
        self.m_regHelper.EvaluateRegistryValue()
        # WriteRegPolResults is called in the helper (full port in next batch when needed)
        self.WriteSecTemplateResults(x_output_doc)

File: CS/PolicyAnalyzer/SharedCode/SecTemplateRegistryItem.cs
using System;
using System.Xml;

namespace SharedCode
{
	// Token: 0x0200003F RID: 63
	internal class SecTemplateRegistryItem : SecTemplateItem
	{
		// Token: 0x060000C2 RID: 194 RVA: 0x0000A380 File Offset: 0x00008580
		public SecTemplateRegistryItem(XmlNode node)
		{
			this.m_regHelper = new RegistryItemHelper(this, "ComputerConfig");
			base.ReadStdElems(node);
			base.ReadStdSecTemplateElems(node);
			string[] array = this.m_sLineItem.Split(new char[] { this.KeyValSeparator() });
			string text = array[0].Substring(8);
			int num = text.LastIndexOf('\\');
			this.m_regHelper.m_sKey = text.Substring(0, num).Trim();
			this.m_regHelper.m_sValue = text.Substring(num + 1).Trim();
			string[] array2 = array[1].Split(new char[] { ',' }, 2);
			string text2 = array2[0];
			if (!(text2 == "1"))
			{
				if (!(text2 == "2"))
				{
					if (!(text2 == "3"))
					{
						if (!(text2 == "4"))
						{
							if (!(text2 == "7"))
							{
								if (!(text2 == "11"))
								{
									this.m_regHelper.m_sType = "regType " + array2[0];
								}
								else
								{
									this.m_regHelper.m_sType = "REG_QWORD";
								}
							}
							else
							{
								this.m_regHelper.m_sType = "REG_MULTI_SZ";
							}
						}
						else
						{
							this.m_regHelper.m_sType = "REG_DWORD";
						}
					}
					else
					{
						this.m_regHelper.m_sType = "REG_BINARY";
					}
				}
				else
				{
					this.m_regHelper.m_sType = "REG_EXPAND_SZ";
				}
			}
			else
			{
				this.m_regHelper.m_sType = "REG_SZ";
			}
			this.m_sBaselineValue = array2[1].Trim(new char[] { '"' });
		}

		// Token: 0x060000C3 RID: 195 RVA: 0x0000A518 File Offset: 0x00008718
		public override string SortKey()
		{
			return string.Concat(new string[]
			{
				base.GetType().Name,
				"!",
				this.m_regHelper.m_sKey,
				"!",
				this.m_regHelper.m_sValue
			}).ToLower();
		}

		// Token: 0x060000C4 RID: 196 RVA: 0x0000A570 File Offset: 0x00008770
		public override void Evaluate(EffectiveStateSources sources, XmlDocument xOutputDoc)
		{
			this.m_regHelper.RetrieveRegistryValue(sources.hklm64);
			this.m_regHelper.EvaluateRegistryValue();
			this.m_regHelper.WriteRegPolResults(xOutputDoc, string.Empty, Environment.MachineName + " - " + sources.hklm64.Name);
		}

		// Token: 0x0400067F RID: 1663
		private RegistryItemHelper m_regHelper;
	}
}
